lib/broccoli/ember-app.js

```diff
@@ -250,7 +242,21 @@ EmberApp.prototype._initOptions = function(options) {
       vendor: vendorTree,
       public: publicTree
     }
-  }, DEFAULT_CONFIG);
+  };
+
+  var emberCLIBabelInstance = this.project.findAddonByName('ember-cli-babel');
+  if (emberCLIBabelInstance) {
+    // future versions of ember-cli-babel will be moving the location for its
+    // own configuration options out of `babel` and will be issuing a deprecation
+    // if used in the older way
+    //
+    // see: https://github.com/babel/ember-cli-babel/pull/105
+    var emberCLIBabelConfigKey = emberCLIBabelInstance.configKey || 'babel';
+    detectedOptions[emberCLIBabelConfigKey] = detectedOptions[emberCLIBabelConfigKey] || {};
```

Comment:
@nathanhammond - Well, we can always set `detectedOptions['ember-cli-babel'].compileModules = true`, but we **cannot** always set `detectedOptions.babel.compileModules = true` (becuase this is exactly what triggers the deprecation).  The entire point here is that we have to figure out what config location to put ember-cli-babel's own config (defaulting to `babel` for backwards compat).